public/app/features/dashboard/components/DashNav/DashNav.tsx

```diff
@@ -75,6 +77,9 @@ export const DashNav = React.memo<Props>((props) => {
   const forceUpdate = useForceUpdate();
   const { chrome } = useGrafana();
 
+  // We don't really care about the event payload here only that it triggeres a re-render of this component
+  useBusEvent(props.dashboard.events, DashboardMetaChangedEvent);
```

Comment:
Why do we use an event to manually determine when to rerender the component? Once the dashboard meta gets updated, won't that trigger a rerender for everything that uses it in the component tree?

Reply:
@owensmallwood as I said the dashboard model is very old an immutable, updating dashboard.meta  does not create a new instance of dashboard and re-trigger a render of DashboardPage and then DashNav. We have a single dashboard class instance (of DashboardModel) that is there for the life of the dashboard. Not great from a react update perspective hence the different events that react components can use to subscribe for changes (Like DashboardGrid will listen for panel change event). 

This will be fixed in the new dashboard runtime

Reply:
It's basically the same as useSelector (that also is subscription under the good for a state change), but here it's an event instead